src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Utility/SafeJsonConvertWrapper.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+namespace Microsoft.Azure.Management.ResourceManager
+{
+    using Microsoft.Azure.Management.ResourceManager.Models;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    public static class SafeJsonConvert
+    {
+        public static string SerializeObject(object obj, JsonSerializerSettings settings)
+        {
+            if (obj != null && obj is Deployment)
+            {
+                Deployment deployment = (Deployment)obj;
+
+                if (deployment.Properties != null)
+                {
+                    if (deployment.Properties.Template is string)
+                    {
+                        deployment.Properties.Template = JObject.Parse((string)deployment.Properties.Template);
```

Comment:
Good point. Right now we throw an exception at "Newtonsoft.Json.JsonTextReader.ParseProperty()". We can catch it and throw a Microsoft.Rest.SerializationException.